File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/config/GpsMockConfig.java
package com.didichuxing.doraemonkit.config;

import com.didichuxing.doraemonkit.constant.CachesKey;
import com.didichuxing.doraemonkit.constant.SharedPrefsKey;
import com.didichuxing.doraemonkit.model.LatLng;
import com.didichuxing.doraemonkit.util.DoKitCacheUtils;
import com.didichuxing.doraemonkit.util.DoKitSPUtil;

/**
 * Created by wanglikun on 2018/9/20.
 */

public class GpsMockConfig {
    /**
     * @return
     */
    public static boolean isGPSMockOpen() {
        return DoKitSPUtil.getBoolean(SharedPrefsKey.GPS_MOCK_OPEN, false);
    }

    /**
     * @param open
     */
    public static void setGPSMockOpen(boolean open) {
        DoKitSPUtil.putBoolean(SharedPrefsKey.GPS_MOCK_OPEN, open);
    }

    public static LatLng getMockLocation() {
        return (LatLng) DoKitCacheUtils.readObject(CachesKey.MOCK_LOCATION);
    }

    public static void saveMockLocation(LatLng latLng) {
        DoKitCacheUtils.saveObject(CachesKey.MOCK_LOCATION, latLng);
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/config/GpsMockConfig.kt
package com.didichuxing.doraemonkit.config

import com.didichuxing.doraemonkit.constant.CachesKey
import com.didichuxing.doraemonkit.constant.SharedPrefsKey
import com.didichuxing.doraemonkit.model.LatLng
import com.didichuxing.doraemonkit.util.CacheUtils
import com.didichuxing.doraemonkit.util.SharedPrefsUtil

/**
 * @author lostjobs created on 2020/6/27
 */
object GpsMockConfig {
    fun isGPSMockOpen() = SharedPrefsUtil.getBoolean(SharedPrefsKey.GPS_MOCK_OPEN, false)
    fun setGPSMockOpen(open: Boolean) = SharedPrefsUtil.putBoolean(SharedPrefsKey.GPS_MOCK_OPEN, open)
    fun getMockLocation(): LatLng? = CacheUtils.readObject(CachesKey.MOCK_LOCATION) as LatLng?
    fun saveMockLocation(latLng: LatLng?) = CacheUtils.saveObject(CachesKey.MOCK_LOCATION, latLng)
}